### PR TITLE
Add Download concurrency to remote form

### PR DESCRIPTION
**Pull request by ZitaNemeckova**
Closes-Issue: AAH-211

Before:
Screenshot 2020-12-10 at 13 41 44

After:
Screenshot 2020-12-10 at 13 37 38
Screenshot 2020-12-10 at 15 27 43

### CODE REVIEW

**Review by chouseknecht — DISMISSED** [review]

File: src/components/repositories/remote-form.tsx [hunk 1]
```diff
@@ -435,6 +435,19 @@ export class RemoteForm extends React.Component<IProps, IState> {
...
             </Flex>
           </FormGroup>
+          <FormGroup
+            fieldId={'download_concurrency'}
+            label={'Download concurrency'}
+          >
+            <TextInput
+              id='download_concurrency'
+              type='number'
+              value={remote.download_concurrency || 0}
```
Comment (chouseknecht) [inline src/components/repositories/remote-form.tsx:445]: Should this default to 1 ?

According to https://docs.pulpproject.org/pulp_ansible/restapi.html#operation/remotes_ansible_collection_create, the value should be >= 1.

Comment (newswangerd) [reply src/components/repositories/remote-form.tsx:445]: @chouseknecht this gets created by the installer. If you want it to default to anything other than 10, it needs to happen when it's created.

@ZitaNemeckova can we default it to "" or null, or does react complain about the element being uncontrolled if you do that?

Comment (ZitaNemeckova) [reply src/components/repositories/remote-form.tsx:445]: Removing default value and adding error message when a user tries to set it below 1.

**Review by newswangerd — APPROVED** [review]

**Comment by newswangerd** [comment]
LGTM

File: src/components/repositories/remote-form.tsx [hunk 2]
```diff
@@ -435,6 +435,22 @@ export class RemoteForm extends React.Component<IProps, IState> {
               </FlexItem>
             </Flex>
           </FormGroup>
+          <FormGroup
+            fieldId={'download_concurrency'}
+            label={'Download concurrency'}
+            validated={remote.download_concurrency > 0 ? 'default' : 'error'}
+            helperTextInvalid={'Number must be bigger than 0'}
```
Comment (chouseknecht) [inline src/components/repositories/remote-form.tsx:442]: This is a little nit-picky. Sorry. Can we make this say, "Value must be greater than 0"

Comment (ZitaNemeckova) [reply src/components/repositories/remote-form.tsx]: Fixed :)

**Review by chouseknecht — APPROVED** [review]

**Pull request merged**